src/main/java/org/fungover/thunder/Main.java

```diff
@@ -1,16 +1,19 @@
 package org.fungover.thunder;
 
 import java.io.IOException;
+import java.net.ServerSocket;
 
 public class Main {
 
-    public static void main(String[] args){
-        Server server;
-        try {
-            server = new Server();
-            server.start();
-        }  catch (IOException | NullPointerException e) {
-            e.printStackTrace();
-        }
+    public static void main(String[] args) throws IOException {
+        ServerSocket serverSocket = new ServerSocket(1883);
+        ClientHandler clientHandler = new ClientHandler();
+        Broker broker = new Broker(clientHandler, serverSocket);
+        createAndStartBroker(broker);
     }
+
+    public static void createAndStartBroker(Broker broker) throws IOException {
+        broker.start();
```

Comment:
I do not understand this refactor. I feel it became more complex than before but seem to use same logic?

Reply:
I truly understand that. The only reason I refactored was to be able to test the method and get the test coverage up to 80 % 😄😄😄😄